pymongo/lock.py

```diff
@@ -11,15 +11,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+"""Internal helpers for lock and condition coordination primitives."""
+
 from __future__ import annotations
 
 import asyncio
-import collections
 import os
+import sys
 import threading
-import time
 import weakref
-from typing import Any, Callable, Optional, TypeVar
+from asyncio import TimeoutError, wait_for
```

Comment:
Suggest using `asyncio.TimeoutError` to avoid shadowing the builtin TimeoutError.